### ISSUE TITLE
ACCOUNTING: Implement accountant billing view

### ISSUE THREAD

**Issue by colinxfleming** [DCAFEngineering/dcaf_case_management#858]
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**
Generate a billing report for accountants!

* **What feature or behavior is this required for?**
Easier accountant reconciliation data entry, since we tend to punch these in ~50 at a time!

* **How should we solve this issue?**
@eheintzelman has put together a slamming design, so we're mostly just writing backend and frontend code to match!

![accountant billing report - annotated](https://cloud.githubusercontent.com/assets/17989540/21289345/43d3ba04-c469-11e6-92d8-ff22f8b44660.png)

I think what we want here is:
* a `Patient` method that filters down to patients who have been marked as `fulfilled` (see the `Fulfillment` object)
* Revise the accountant table_content to match the designs, with a few changes: 
  * make abortion care $, weeks at proc, and pledge sent date editable
* On loading the accountants page, make the report just load for now. don't worry about the `reports` dropdown above the accountant billing section
* keep the search thing functioning the way it is now
* add an integration test to confirm that you can find a patient from this search bar, and edit a patient's fulfillment information (we can spin this off if needed)

The troublesome parts of this are probably more frontend than backend, so marking accordingly.

Stems from #643

**Comment by lomky** [comment]
Tagging interest on this.

**Comment by lomky** [comment]
Working on this for hacknight

**Comment by lomky** [comment]
I have a bit done on this, but nothing substantial. Planning on picking it up again at hacknight Monday, but if someone wants to knock it out before then they're welcome to steal it :)

**Comment by FeminismIsAwesome** [comment]
I'm gonna steal zing! @lomky do you have any code/branch that I might want to look at or should I go at it from scratch?

**Comment by lomky** [comment]
Thank you for your theft @FeminismIsAwesome !

Here is my scratch work I put together, it's not much yet:

https://github.com/lomky/dcaf_case_management/compare/master...lomky:858_implement_accountant_billing_view

**Comment by colinxfleming** [comment]
moving from the slack channel because I wrote kinda unclear directions here (sorry!):

* the code should live in `app/views/accountants/index.html.erb` and associated partials
* do the fields as follows (overruling what's in the designs):
- clinic (sourced from `patient.clinic_name` / `patient.clinic.name`)
- pt initials (source from `patient.name`)
- abortion care $ (`patient.fulfillment.procedure_cost`, input field)
- weeks at procedure (`patient.fulfillment.gestation_at_procedure`, input field)
- pledge sent date (no great way to do this right now. let's use a placeholder for now and make a note to wire up something.)
- check number (`patient.fulfillment.check_number`, input field)
- date of check (`patient.fulfillment.date_of_check`)

**Comment by colinxfleming** [comment]
Now that we can assign stuff I'm gonna assign to you @FeminismIsAwesome ! Please pop up a WIP PR if you have anything written when you get a chance!

**Comment by tingaloo** [comment]
Took a stab at this, but one thing I'm stumped on is iterating through Patients that have a fulfillment object that exists.

My mongoid isn't that fresh but I've tried variations of these
```
@test = Patient.where('fulfillment.created_at'.ne => nil)
 @test = Patient.where(:'fulfillment'.exists => true)
```

Could use some mongoid expertise on this one.

**Comment by colinxfleming** [comment]
We do this kinda stupidly and create a fulfillment object at point of patient creation. Which is an antipattern we should do away with at some point! But I think in the meantime the best way to get at fulfille patients is like so:

```
Patient.where('fulfillment.fulfilled' => true)
```

To get at nested fields, easiest to wrap em in quotes and access fields with dots, I think.

**Comment by colinxfleming** [comment]
Closed via #1292
